orangecontrib/prototypes/widgets/owinteractions.py

```diff
@@ -82,27 +107,108 @@ def get_states(self, initial_state):
 		return states
 
 
+class InteractionItemDelegate(gui.TableBarItem):
+	def __init__(self, **kwargs):
+		super().__init__(**kwargs)
+		self.r = QColor(255, 170, 127)
+		self.g = QColor(170, 242, 43)
+		self.b = QColor(70, 190, 250)
+		self.__line = QLineF()
+		self.__pen = QPen(self.b, 5, Qt.SolidLine, Qt.RoundCap)
+
+	def paint(
+			self, painter: QPainter, option: QStyleOptionViewItem,
+			index: QModelIndex
+	) -> None:
+		opt = QStyleOptionViewItem(option)
+		self.initStyleOption(opt, index)
+		widget = option.widget
+		style = QApplication.style() if widget is None else widget.style()
+		self.__style = style
+		text = opt.text
+		opt.text = ""
+		style.drawControl(QStyle.CE_ItemViewItem, opt, painter, widget)
+		textrect = style.subElementRect(
+			QStyle.SE_ItemViewItemText, opt, widget)
+
+		center = self.cachedData(index, InteractionRank.IntRole)
+		if center is not None:
```

Comment:
Isn't in this case `center` always defined? If so, why the condition?

Reply:
center is only defined for the leftmost column. for the other two columns it is `None` so that the drawing is skipped and the text still appears.

Reply:
Makes sense. That would make a great comment in code. Anytime you use assumptions that are likely unclear to someone reading the code, please make a comment.

Reply:
I think you explained it much better on Github than in the code: here you also provided some context why would only interaction items draw the bars, there that is left as a question to the poor programmer maintaining this code after a few years... I would appreciate if you expand this comment, but will not block the merge because of it.